### PR TITLE
BXMSDOC-1433 (for fork "final"): New "Guided Decision Table Graphs" (i.e., linked tables) section in BRMS User Guide

### PR DESCRIPTION
**Pull request by sterobin**
Added new "Guided Decision Table Graphs" section in BRMS User Guide (sec 8.5).

Links:
- [JIRA](https://issues.jboss.org/browse/BXMSDOC-1433)
- [Rendered output](http://file.rdu.redhat.com/~sterobin/BXMSDOC-1433/#guided_decision_table_graphs_con)

PR is for 1433-master to 1433-master-final within fork, in preparation for upstream merge.

### CODE REVIEW

File: docs/product-user-guide/src/main/asciidoc/chap-writing-rules.adoc [hunk 1]
```diff
@@ -330,6 +330,145 @@ include::guided-decision-tables-columns-edit-proc.adoc[leveloffset=+2]
...
+
+
+== Guided Decision Trees
+
+A decision tree is a graphical representation of a decision model in a tree-like manner. You can create simple decision trees in Business Central with flat data object models. The editor does not support nested data objects.
+
+The editor offers a palette on the left-hand side (with available data objects, fields and corresponding actions) and a working area on the right-hand side where you can drag and drop the data objects to build a decision tree.
+You can use connectors and applicable child objects prompted by the editor, to complete your tree. You can manipulate each node using its `Delete`, `Edit`, and `Collapse` icons.
+
+While creating a decision tree, you must remember the following points:
```
Comment (vidya-iyengar) [inline docs/product-user-guide/src/main/asciidoc/chap-writing-rules.adoc]: Can do without "you must " in this sentence.

File: docs/product-user-guide/src/main/asciidoc/chap-writing-rules.adoc [hunk 2]
```diff
@@ -330,6 +330,145 @@ include::guided-decision-tables-columns-edit-proc.adoc[leveloffset=+2]
...
+
+
+Business Central provides verification and validation feature to help ensure that a guided decision table is complete and error free.
+The feature looks for gaps in the logic of a guided decision table and validates the relationship between different cells.
+Most of the issues that the Business Central verification and validation feature reports are gaps in the author's logic.
+The verification and validation feature does not prevent you from saving your work if you choose to ignore the verification and validation notifications.
+
+When you edit a guided decision table, the verification and validation feature notifies you if you do something wrong.
+For example, if you forget to set an action for a row of the guided decision table or if you have a duplicate row, a new panel, Analysis, will open in the Business Central with a notification about the issue.
+The Analysis panel presents a list of issues found in the guided decision table, each item also pointing out the affected guided decision table rows.
```
Comment (vidya-iyengar) [inline docs/product-user-guide/src/main/asciidoc/chap-writing-rules.adoc:381]: "Analysis" is a UI panel, so it should be bold right?

File: docs/product-user-guide/src/main/asciidoc/chap-writing-rules.adoc [hunk 3]
```diff
@@ -330,6 +330,145 @@ include::guided-decision-tables-columns-edit-proc.adoc[leveloffset=+2]
...
++
+In this case, there is no way to know if the loan will be approved or not.
++
+The example below illustrates conflict between two cells in a decision table:
+
+** When Age > 25
+** When Age < 25
+
++
+A row with conflicting cells never execute.
```
Comment (vidya-iyengar) [inline docs/product-user-guide/src/main/asciidoc/chap-writing-rules.adoc]: execute > executes.

File: docs/product-user-guide/src/main/asciidoc/chap-writing-rules.adoc [hunk 4]
```diff
@@ -330,6 +330,145 @@ include::guided-decision-tables-columns-edit-proc.adoc[leveloffset=+2]
...
+
+** when Age > 20 then Approve Loan
+** when Deposit < 20000 then Refuse Loan
+
++
+These two rules may lead to confusion for a person who is over 20 years old and have deposited less than 20000.
+You may add more constraints to avoid the conflict after noticing the warning to make your rule logic complete.
+* Missing Columns
++
+In some cases, usually by accident, the user can delete all the condition or action columns.
```
Comment (vidya-iyengar) [inline docs/product-user-guide/src/main/asciidoc/chap-writing-rules.adoc:434]: In some cases, usually by accident, a user can ...

File: docs/product-user-guide/src/main/asciidoc/chap-writing-rules.adoc [hunk 5]
```diff
@@ -330,6 +330,145 @@ include::guided-decision-tables-columns-edit-proc.adoc[leveloffset=+2]
...
+When the conditions are removed all the actions are executed and when the actions columns are missing the rows do nothing.
```
Comment (vidya-iyengar) [inline docs/product-user-guide/src/main/asciidoc/chap-writing-rules.adoc]: Missing comma after `removed` and 'missing' in: "When the conditions are removed, all the actions..."

File: docs/product-user-guide/src/main/asciidoc/chap-writing-rules.adoc [hunk 6]
```diff
@@ -330,6 +330,145 @@ include::guided-decision-tables-columns-edit-proc.adoc[leveloffset=+2]
...
+Both may or may not be by design, usually though this is a mistake.
+
+
+=== Reporting
+
+
+The verification and validation feature of Business Central reports different issue levels in the Analysis panel while you are updating a guided decision table.
```
Comment (vidya-iyengar) [inline docs/product-user-guide/src/main/asciidoc/chap-writing-rules.adoc:442]: Analysis panel : bold

File: docs/product-user-guide/src/main/asciidoc/chap-writing-rules.adoc [hunk 7]
```diff
@@ -330,6 +330,145 @@ include::guided-decision-tables-columns-edit-proc.adoc[leveloffset=+2]
...
+
+* image:gdtValidationVerificationIconError.png[] Error: This means a serious fault, which may lead to the guided decision table failing to work as designed at run time. Conflicts, for example, are reported as errors.
+* image:gdtValidationVerificationIconWarning.png[] Warning: This is most likely a serious fault, however they may not prevent the guided decision table from working but need to be double checked. Warnings are used e.g. for subsumption.
+* image:gdtValidationVerificationIconInfo.png[] Information: This kind of issues might be a design decision of the author of the guided decision table as well as a simple accident. This is used e.g. for a row missing any condition.
```
Comment (vidya-iyengar) [inline docs/product-user-guide/src/main/asciidoc/chap-writing-rules.adoc:446]: Replace e.g. with "for example"

File: docs/product-user-guide/src/main/asciidoc/guided-decision-table-graphs-con.adoc [hunk 8]
```diff
@@ -0,0 +1,14 @@
+[#guided_decision_table_graphs_con]
+= Guided Decision Table Graphs
+
+A Guided Decision Table Graph is a collection of related guided decision tables that are displayed within a single editor. You can use this editor to better visualize and work with various related decision tables in one location. Additionally, when a condition or an action in one table uses the same data type as a condition or an action in another table, the tables will be physically linked with a line in the table graph editor.
+
+For example, if one decision table determines a loan application rate and another table uses the application rate to determine some other action, then the two decision tables are linked in a guided decision table graph.
```
Comment (vidya-iyengar) [inline docs/product-user-guide/src/main/asciidoc/guided-decision-table-graphs-con.adoc:6]: then the two decision tables are linked in a guided decision table graph >>
"then both the decision tables are linked in a guided decision table graph."

File: docs/product-user-guide/src/main/asciidoc/guided-decision-table-graphs-remove-tables-proc.adoc [hunk 9]
```diff
@@ -0,0 +1,16 @@
+[#guided_decision_table_graphs_remove_tables_proc]
+= Removing Decision Tables from a Guided Decision Table Graph
+
+You can remove decision tables from the guided decision table graph editor without deleting the decision table file itself.
```
Comment (vidya-iyengar) [inline docs/product-user-guide/src/main/asciidoc/guided-decision-table-graphs-remove-tables-proc.adoc]: can do without the word "itself"

File: docs/product-user-guide/src/main/asciidoc/guided-decision-table-graphs-con.adoc (hunk 8 above)
Comment (sterobin) [reply docs/product-user-guide/src/main/asciidoc/guided-decision-table-graphs-con.adoc:6]: I'd actually argue that it's better to say "the two graphs are linked" vs. "both graphs are linked" in this case. To me, "both" sounds comprehensive, like "all" but for when there are just two options, often used emphatically. But table graphs can have many decision tables. Two could be linked, three others could be linked, and two others linked. So three separate instances of linked tables within a graph, for instance. I'm trying to just say that these two would be linked, not to say that others necessarily will be, if there are any. Not sure if I'm explaining myself well, but either way, I think "two" is better than "both" here.

**Pull request merged**